src/main/scala/is/hail/io/bgen/BgenRecord.scala

```diff
@@ -78,6 +78,73 @@ class BgenRecordV11(compressed: Boolean,
   }
 }
 
+class BGen12ProbabilityArray(a: Array[Byte], nSamples: Int, nGenotypes: Int, nBitsPerProb: Int) {
+
+  def apply(s: Int, i: Int): UInt = {
+    val firstBit = (s * (nGenotypes - 1) + i) * nBitsPerProb
+    val firstBitWithinByte = firstBit & 7
+
+    var j = nSamples + 10 + (firstBit >> 3)
```

Comment:
I don't think it matters.  If firstByte is negative something really bad is happening.